### DISCUSSION TITLE
fail2ban regex

### DISCUSSION THREAD

**Discussion by phonesrescue** [fail2ban/fail2ban#3906] [General]
I can’t manage to set up the regex for Fail2Ban to block something like this:

```
54.88.84.219 - - [21/Dec/2024:13:41:37 +0000] "GET /events/category/stand-up/day/4586-04-01/ HTTP/1.1" 200 46581 "-" "Mozilla/5.0 AppleWebKit/537.36 (KHTML, like Gecko; compatible; Amazonbot/0.1; +https://developer.amazon.com/support/amazonbot) Chrome/119.0.6045.214 Safari/537.36"
``` 

I tried: 
```
failregex = ^<HOST> - - \[\d{2}/[A-Za-z]{3}/\d{4}:\d{2}:\d{2}:\d{2} [\+\-]\d{4}\] "GET /.*" 200 \d+ "-" "Mozilla/5.0 AppleWebKit/537.36 \(KHTML, like Gecko; compatible; Amazonbot/0.1.*\)" 
```
But for some reason it does not work!

**Comment by sebres** [comment]
> But for some reason it does not work!

Without to investigate deeper - you have not to cover part of string matching the datepattern (timestamp in square brackets) - as the documentation says, fail2ban will cut it from the message before applies failregex.

See 

Try something like this:
```ini
badbots = (?:Amazonbot/0.1)
failregex = ^<ADDR>[^"]*"[^"]*" \d+ \d+ "[^"]*" "[^"]*\b%(badbots)s\b
```
Also consider https://github.com/fail2ban/fail2ban/discussions/3772#discussioncomment-9870391 ...

**Reply by phonesrescue** [reply]
I found the cause. My server is OpenLiteSpeed with CyberPanel. While the main web server logs in the `access.log` file are recorded in the format: `%h %l %u %t "%r" %>s %b "%{Referer}i" "%{User-Agent}i`, the logs for individual sites in the `/home` directory are written in the format: `"%h %l %u %t "%r" %>s %b "%{Referer}i" "%{User-Agent}i"` (with additional quotation marks at the beginning and end). I have no idea why this is the case – whether it's a feature of OpenLiteSpeed or CyberPanel. The problem is solved. Thanks!
